src/main/java/frc/robot/robot_manager/RobotManager.java

```diff
@@ -49,102 +49,90 @@ protected void collectInputs() {}
+    return switch (currentState) {
       case SPEAKER_WAITING,
-          AMP_WAITING,
-          FEEDING_WAITING,
-          PASS_WAITING,
-          IDLE_NO_GP,
-          IDLE_WITH_GP,
-          CLIMBING_1_LINEUP,
-          CLIMBING_2_HANGING -> {}
-      case SPEAKER_SCORING -> {
-        if (!queuer.hasNote()) {
-          setStateFromRequest(RobotState.IDLE_NO_GP);
-        }
-      }
-      case AMP_SCORING -> {
-        if (!queuer.hasNote()) {
-          setStateFromRequest(RobotState.IDLE_NO_GP);
-        }
-      }
-      case FEEDING_SHOOTING -> {
-        if (!queuer.hasNote()) {
-          setStateFromRequest(RobotState.IDLE_NO_GP);
-        }
-      }
-      case PASS_SHOOTING -> {
-        if (!queuer.hasNote()) {
-          setStateFromRequest(RobotState.IDLE_NO_GP);
-        }
-      }
-      case SPEAKER_PREPARE_TO_SCORE -> {
-        if (shooter.atGoal() && arm.atGoal()) {
-          setStateFromRequest(RobotState.SPEAKER_SCORING);
-        }
-      }
+              AMP_WAITING,
+              FEEDING_WAITING,
+              PASS_WAITING,
+              IDLE_NO_GP,
+              IDLE_WITH_GP,
+              CLIMBING_1_LINEUP,
+              CLIMBING_2_HANGING ->
+          currentState;
+      case SPEAKER_SCORING, AMP_SCORING, FEEDING_SHOOTING, PASS_SHOOTING ->
+          !queuer.hasNote() ? RobotState.IDLE_NO_GP : currentState;
 
-      case AMP_PREPARE_TO_SCORE -> {
-        if (shooter.atGoal() && arm.atGoal()) {
-          setStateFromRequest(RobotState.AMP_SCORING);
-        }
-      }
-      case FEEDING_PREPARE_TO_SHOOT -> {
-        if (shooter.atGoal() && arm.atGoal()) {
-          setStateFromRequest(RobotState.FEEDING_SHOOTING);
-        }
-      }
-      case PASS_PREPARE_TO_SHOOT -> {
-        if (shooter.atGoal() && arm.atGoal()) {
-          setStateFromRequest(RobotState.PASS_SHOOTING);
-        }
-      }
-      case UNJAM -> {}
-      case INTAKING -> {
-        if (intake.hasNote() || queuer.hasNote()) {
-          setStateFromRequest(RobotState.IDLE_WITH_GP);
-        }
-      }
-      case OUTTAKING -> {
-        if (!intake.hasNote() && !queuer.hasNote()) {
-          setStateFromRequest(RobotState.IDLE_NO_GP);
-        }
-      }
-    }
+      case SPEAKER_PREPARE_TO_SCORE ->
+          shooter.atGoal() && arm.atGoal() ? RobotState.SPEAKER_SCORING : currentState;
+
+      case AMP_PREPARE_TO_SCORE ->
+          shooter.atGoal() && arm.atGoal() ? RobotState.AMP_SCORING : currentState;
 
-    return currentState;
+      case FEEDING_PREPARE_TO_SHOOT ->
+          shooter.atGoal() && arm.atGoal() ? RobotState.FEEDING_SHOOTING : currentState;
+      case PASS_PREPARE_TO_SHOOT ->
+          shooter.atGoal() && arm.atGoal() ? RobotState.PASS_SHOOTING : currentState;
+      case UNJAM -> currentState;
+      case INTAKING -> queuer.hasNote() ? RobotState.IDLE_WITH_GP : currentState;
+      case OUTTAKING ->
+          !queuer.hasNote() || !intake.hasNote() ? RobotState.IDLE_NO_GP : currentState;
```

Comment:
this logic is correct but i think it could also be written like this, which is a lil simpler:
```suggestion
          queuer.hasNote() || intake.hasNote() ? currentState : RobotState.IDLE_NO_GP;
```
"if the queuer or intake has a note, keep outtaking, otherwise IDLE_NO_GP"